### PR TITLE
feat(bind OverWindow): rewrite Avg and other complex AggKind in OverWindow

### LINKED ISSUE #9929: feat(bind overWindow): rewrite Avg and other complex aggKind in overWindow
The binder should rewrite AVG, StddevPop and other complex calls into multiple aggCall and project with expressions.

https://github.com/risingwavelabs/risingwave/blob/237ed2e38719454722864ed6d05b43d4cab27855/src/frontend/src/optimizer/plan_node/logical_agg.rs#L499-L543

### PR DESCRIPTION
**Pull request by Honeta**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

close #9929

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/frontend/src/expr/window_function.rs [hunk 1]
```diff
@@ -124,7 +124,7 @@ impl std::fmt::Debug for WindowFunction {
             }
             builder.finish()
         } else {
-            write!(f, "{}() OVER(", self.kind)?;
+            write!(f, "{}({:?}) OVER(", self.kind, self.args)?;
```
Comment (richardchien) [inline src/frontend/src/expr/window_function.rs:127]: What about display arguments line in `PlanWindowFunctionDisplay`?

https://github.com/risingwavelabs/risingwave/blob/67ab16bc058397fda7f1c7bee6dfeb99480cde7a/src/frontend/src/optimizer/plan_node/generic/over_window.rs#L60-L75

File: src/frontend/src/optimizer/plan_node/logical_over_window.rs [hunk 2]
```diff
@@ -98,14 +120,214 @@ impl LogicalOverWindow {
...
+                    && matches!(
+                        agg_kind,
+                        AggKind::Avg
+                            | AggKind::StddevPop
+                            | AggKind::StddevSamp
+                            | AggKind::VarPop
+                            | AggKind::VarSamp
+                    )
+                {
+                    // Refer to LogicalAggBuilder::try_rewrite_agg_call()
```
Comment (richardchien) [inline src/frontend/src/optimizer/plan_node/logical_over_window.rs:142]: Could we extract the common logic into something like `Agg::breakdown_agg_call`?

Comment (st1page) [reply src/frontend/src/optimizer/plan_node/logical_over_window.rs:142]: I think we can do it in later PR. It is a little comples

**Review by richardchien — APPROVED** [review]
LGTM!

**Pull request merged**